src/v5.rs

```diff
@@ -79,11 +79,13 @@ fn write_addr(mut packet: &mut [u8], target: &TargetAddr) -> io::Result<usize> {
         }
         TargetAddr::Domain(ref domain, port) => {
             packet.write_u8(3).unwrap();
-            if domain.len() > u8::max_value() as usize {
+            let domain_bytes = domain.as_bytes();
```

Comment:
I'd revert these additional variable assignments again now that it is clear that `domain.len()` and `domain.as_bytes().len()` are equivalent. IMO, the code is easier to read that way.

Reply:
Oh, is it? Personally I was deceived into believing the opposite. Maybe it's just that I haven't used rust for several month.

Reply:
@ratijas , I create a code snipped showing counting bytes vs characters.

https://play.rust-lang.org/?gist=459c4d88c27e20ffd4c8c52df7d8e79d&version=stable

Reply:
Yep, I'd revert the changes here, but the rest of the PR looks good!